File: Project/alarm/src-gen/alarm/State.java
/**
 */
package alarm;

import org.eclipse.emf.common.util.EList;

/**
 * <!-- begin-user-doc -->
 * A representation of the model object '<em><b>State</b></em>'.
 * <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link alarm.State#getActions <em>Actions</em>}</li>
 *   <li>{@link alarm.State#getTransition <em>Transition</em>}</li>
 * </ul>
 *
 * @see alarm.alarmPackage#getState()
 * @model
 * @generated
 */
public interface State extends NamedElement {
	/**
	 * Returns the value of the '<em><b>Actions</b></em>' containment reference list.
	 * The list contents are of type {@link alarm.Action}.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Actions</em>' containment reference list.
	 * @see alarm.alarmPackage#getState_Actions()
	 * @model containment="true" required="true"
	 * @generated
	 */
	EList<Action> getActions();

	/**
	 * Returns the value of the '<em><b>Transition</b></em>' containment reference.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Transition</em>' containment reference.
	 * @see #setTransition(Transition)
	 * @see alarm.alarmPackage#getState_Transition()
	 * @model containment="true" required="true"
	 * @generated
	 */
	Transition getTransition();

	/**
	 * Sets the value of the '{@link alarm.State#getTransition <em>Transition</em>}' containment reference.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Transition</em>' containment reference.
	 * @see #getTransition()
	 * @generated
	 */
	void setTransition(Transition value);

	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @model
	 * @generated
	 */
	void accept(Visitor visitor);

} // State

File: ProjectJS/src/kernel/model/State.ts
import Action from "./Action";
import AppVisitor from "./utils/AppVisitor";
import NamedElement from "./NamedElement";
import Sensor from "./Sensor";
import Transition from "./Transition";
import Visitable from "./utils/Visitable";
import SensorTransition from "./SensorTransition";

export default class State extends NamedElement implements Visitable{
    private _actions: Action[];
    private _transitions: Transition[];

    constructor(name:string,transitions:Transition[],actions:Action[]){
        super(name);
        this._transitions = transitions;
        this._actions = actions;
    }


    /**
     * Getter actions
     * @return {Action[]}
     */
	public get actions(): Action[] {
		return this._actions;
	}

    /**
     * Getter transitions
     * @return {Transition[]}
     */
	public get transitions(): Transition[] {
		return this._transitions;
	}

    /**
     * Setter actions
     * @param {Action[]} value
     */
	public set actions(value: Action[]) {
		this._actions = value;
	}

    /**
     * Setter transitions
     * @param {Transition[]} value
     */
	public set transitions(value: Transition[]) {
		this._transitions = value;
	}
    
    accept(visitor: AppVisitor) {
        return visitor.visitState(this)    
    }

}